### PR TITLE
boot: implement SetNextBoot in terms of bootState.setNext

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

File: boot/bootstate16.go [hunk 1]
```diff
@@ -150,3 +150,38 @@ func (s *bootState16) markSuccessful(update bootStateUpdate) (bootStateUpdate, e
...
+	}
+
+	env := u16.env
+	toCommit := u16.toCommit
+
+	snapMode := "try"
+	rebootRequired = true
+	if env[goodBootVar] == nextBoot {
+		// If we were in anything but default ("") mode before
+		// and now switch to the good core/kernel again, make
```
Comment (anonymouse64) [inline boot/bootstate16.go]: the `now` makes this statement a little hard to parse

```suggestion
		// and switched to the good core/kernel again, make
```

File: boot/boot.go [hunk 2]
```diff
@@ -91,7 +91,12 @@ type Device interface {
 // always be NOP).
 func Participant(s snap.PlaceInfo, t snap.Type, dev Device) BootParticipant {
 	if applicable(s, t, dev) {
-		return &coreBootParticipant{s: s, t: t}
+		bs, err := bootStateFor(t, dev)
+		if err != nil {
+			// all internal errors at this point
+			panic(err)
```
Comment (mvo5) [inline boot/boot.go]: I'm a bit uneasy about this panic, I know it's internal but I wonder if a bug in bootStateFor or something unexpected on the system may put us into a situation where we try to run a task like refresh, try to update the boot, panic, restart and panic again, i.e. into a situation where we cannot recover from via a refresh. Or am I too paranoid here?

Comment (pedronis) [reply boot/boot.go]: the invariant here is that for any t and dev for which applicable is true bootStateFor should not error, this should be well covered between unit tests and spread already.
